modules/basegl/src/processors/splitimage.cpp

```diff
@@ -181,9 +177,14 @@ void SplitImage::process() {
                 shader = SharedOpenGLResources::getPtr()->getNoiseShader();
             }
 
+            glEnable(GL_SCISSOR_TEST);
+            utilgl::ScissorState scissorviewport(viewport);
+
             shader->activate();
             utilgl::singleDrawImagePlaneRect();
             shader->deactivate();
+
+            glDisable(GL_SCISSOR_TEST);
```

Comment:
If the shader calls would throw, the GL state will be left in the wrong state...